k-distribution/k-tutorial/2_intermediate/01_macros/README.md

```diff
@@ -4,6 +4,151 @@ copyright: Copyright (c) Runtime Verification, Inc. All Rights Reserved.
+important that rules be defined correctly to always terminate, otherwise the
+macro expansion phase will run forever. In practice, the types of macros
+usually written are very easy to ensure this property on, however.
+
+### Exercise
+
+Using a `Nat` sort containing the constructors `0` and `S` (i.e., a
+[Peano-style](https://en.wikipedia.org/wiki/Peano_axioms) axiomatization of the
+natural numbers where `S(N) = N + 1`, `S(S(N)) = N + 2`, etc), write a macro
+that will compute the sum of two numbers.
+
+## Aliases
+
+**NOTE**: future lessons will discuss KORE, an intermediate language for K,
+which is based on Matching Logic. KORE defines a concept called an "alias"
+which is similar, but not exactly the same as the concept we are about to
+introduce, which is a particular variant of a macro. It's worth remembering
+to keep these two concepts separate in your mind.
+
+Macros can be very useful in helping you define a programming language.
+However, they can be disruptive while pretty printing a configuration. For
+example, you might write a set of macros that transforms the code the user
+wrote into equivalent code that is slightly harder to read. This can make it
+more difficult to understand the code when it is pretty printed as part of the
+output of rewriting.
+
+K defines a relatively straightforward but novel solution to this problem,
+which is known as a K **alias**. An alias in K is very similar to a macro,
+with the exception that the rewrite rule will also be applied **backwards**
+during the pretty-printing process.
+
+It is very simple to make a production be an alias instead of a macro: simply
+use the `alias` or `alias-rec` attributes instead of the `macro` or `macro-rec`
+attributes. For example, if the example involving `if` statements above was
+declared using an alias instead of a macro, the `Stmt` term `if (E) {} else {}`
+would be pretty-printed as `if (E) {}`. This is because during pretty-printing,
+the term participates in another macro-expansion pass. However, this macro
+expansion step will only apply rules with the `alias` or `alias-rec` attribute,
+and, critically, it will reverse the rule by treating the left-hand side as if
+it were the right-hand side, and vice versa.
+
+This can be very useful to allow you to define one construct in terms of
+another while still being able to pretty-print the result as if it were
+the original term in question. This can be especially useful for applications
+of K where we are taking the output of rewriting and attempting to use it as
+a code fragment that we then execute, such as with test generation.
+
+### Exercise
+
+Modify `LESSON-01` above to use an alias instead of a macro and experiment
+with how various terms are pretty-printed by invoking `krun` on them.
+
+## `anywhere` rules
+
+There is one other type of rule that this lesson intends to introduce, which
+is called the **anywhere rule**. An anywhere rule is specified by adding the
```

Comment:
Nitpick: This reads slightly awkward to me. Maybe rephrase as "The last type of rule we will introduce in this lesson is the **anywhere rule**."